File: day1/python/main.py
def main():
    frequency = 0
    calibrations = []
    calibration_missing = True
    calibration_count = 0
    calibration_limit = 300
    while calibration_missing:
        with open("input.txt") as input:
            for change in input:
                frequency += int(change)
                # Second answer is found after iterating for a while...
                if calibration_missing and frequency in calibrations:
                    print(f"Answer to challenge #2: {frequency}")
                    calibration_missing = False
                calibrations.append(frequency)
        # First answer is found after first iteration
        if calibration_count == 0:
            print(f"Answer to challenge #1: {frequency}")
            print(f"Running calibrations until {calibration_limit} to get answer #2, go get some coffee...")
        calibration_count += 1
        # Stop iteration if limit is hit
        if calibration_count == calibration_limit:
            calibration_missing = False
            print ("Ended calibration loop without finding duplicates")

if __name__ == "__main__":
    main()

File: day1/rust/src/main.rs
use std::fs::File;
use std::io::BufRead;
use std::io::BufReader;

fn main() {
    let f = File::open("input.txt").expect("file not found");
    let file = BufReader::new(&f);
    let mut frequency = 0;
    for line in file.lines() {
        let line_result = line.unwrap();
        let change: i32 = line_result.trim().parse().expect("Should be a number!");
        frequency += change;
    }
    println!("Answer to challenge #1: {}", frequency);
}
